dags/analyzer_helper/common/fetch_platforms.py

```diff
@@ -0,0 +1,177 @@
+        self.platform_name = platform_name
+        self.db = self.client[db_name]
+        self.collection = self.db[collection]
+
+    def fetch_all(self):
+        """
+        Fetches all Discord platforms from the MongoDB collection.
+
+        Returns:
+            list: A list of dictionaries, each containing platform data with the following fields:
+                - platform_id: The platform ID (_id from MongoDB).
+                - metadata: A dictionary containing period, and id.
+                - recompute: A boolean set to False.
+        """
+        query = {
+            "disconnectedAt": None,
+            "platform": self.platform_name,
+        }
+        projection = {
+            "_id": 1,
+            "metadata.period": 1,
+            "metadata.id": 1,
+        }
+
+        cursor = self.collection.find(query, projection)
+        platforms = []
+
+        for doc in cursor:
+            platform_data = {
+                "platform_id": str(doc["_id"]),
+                "metadata": {
+                    "period": doc.get("metadata", {}).get("period", None),
+                    "id": doc.get("metadata", {}).get("id", None),
+                },
+                "recompute": False,
+            }
+            platforms.append(platform_data)
+
+        return platforms
+
+    def fetch_analyzer_parameters(self, platform_id: str):
+        """
+        Fetches the specified Discord platform from the MongoDB collection with additional fields.
+
+        Parameters:
+            platform_id (str): The platform ID to fetch.
+
+        Returns:
+            dict: A dictionary containing the platform data with the following fields:
+                - platform_id: The platform ID (_id from MongoDB).
+                - metadata: A dictionary containing period, action, window, selectedChannels, and id.
+                - recompute: A boolean set to False.
+        """
+        query = {
+            "_id": ObjectId(platform_id),
+            "disconnectedAt": None,
+            "platform": self.platform_name,
+        }
+        # TODO: introduce strategy pattern once we add 'telegram'
+        if self.platform_name == "discord":
+            projection = {
+                "_id": 1,
+                "metadata.period": 1,
+                "metadata.action": 1,
+                "metadata.window": 1,
+                "metadata.selectedChannels": 1,
+                "metadata.id": 1,
+            }
+        else:
+            projection = {
+                "_id": 1,
+                "metadata.period": 1,
+                "metadata.action": 1,
+                "metadata.window": 1,
+                "metadata.resources": 1,
+                "metadata.id": 1,
+            }
+
+        cursor = self.collection.find(query, projection)
+        platforms = []
+
+        for doc in cursor:
+            if self.platform_name == "discord":
+                selected_channels = doc.get("metadata", {}).get(
```

Comment:
please rename the `selected_channels` to `resources`. In future we're going to just use that term.